File: framework/ui/src/io/intino/alexandria/ui/displays/components/Grouping.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.notifiers.GroupingNotifier;
import io.intino.alexandria.ui.displays.notifiers.GroupingToolbarNotifier;

public class Grouping<DN extends GroupingNotifier, B extends Box> extends AbstractGrouping<DN, B> {

    public Grouping(B box) {
        super(box);
    }

    public Grouping<DN, B> pageSize(int pageSize) {
        notifier.refreshPageSize(pageSize);
        return this;
    }

}

File: web-ui/src/displays/components/Grouping.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractGrouping from "../../../gen/displays/components/AbstractGrouping";
import GroupingNotifier from "../../../gen/displays/notifiers/GroupingNotifier";
import GroupingRequester from "../../../gen/displays/requesters/GroupingRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class Grouping extends AbstractGrouping {

	constructor(props) {
		super(props);
		this.notifier = new GroupingNotifier(this);
		this.requester = new GroupingRequester(this);
	};


}

export default withStyles(styles, { withTheme: true })(Grouping);
DisplayFactory.register("Grouping", withStyles(styles, { withTheme: true })(Grouping));